docs/decisions/0009-webhook-based-communication.rst

```diff
@@ -0,0 +1,197 @@
+- **Zero Infrastructure Cost**: No additional cost associated with using a cloud event bus or message queue. Relies on HTTPS and standard cryptographic libraries.
+- **Lightweight and Fast**: Low overhead in both message size and processing latency. Uses minimal resources and fast cryptographic operations.
+- **Custom Retry Logic**: Since its a custom HTTP based solution, we have the flexibility to implement retry mechanism.
+- **No Payload Limits**: Unlike EventBridge (which enforces a ~256 KB size limit), this model allows payload sizes as needed.
+- **Single Sender–Receiver**: The communication is between one sender and one receiver and be enhanced as needed.
+
+Cons
+====
+
+- **Requires Custom Implementation**: Unlike managed services like AWS EventBridge, this solution requires us to build, test, and maintain the retry logic.
+- **Manual Key Management**: Secrets must be managed and rotated manually or via internal tooling, which adds operational overhead and potential for misconfiguration if not handled properly.
+- **No Built-in Delivery Guarantees**: Unlike EventBridge, which guarantees at-least-once delivery with retries, we need to implement our own retry mechanism.
+
+Rejected Alternatives
+*********************
+
+
+1. Open edX Kafka Event Bus
+===========================
+
+The `kafka-event-bus` is an asynchronous event system used across Open edX services, based on the pub/sub model using Django Signals (via `OpenEdxPublicSignals`). It extends internal Django signals to communicate between distributed services.
+
+**Cons**
+
+- **Coupling to 2U Infrastructure**: The Kafka bus is managed through `edx-terraform` under 2U ownership, introducing infra and org-level dependencies.
+- **Tied to Open edX Events**: Requires all events to be defined in the `openedx-events` repo, adding further tight coupling.
+- **Dependent on OpenEdxPublicSignals**: Built around `OpenEdxPublicSignals`, which assumes deeper integration into Open edX internals.
+- **Django-Specific**: Primarily designed for Django-based services, which is limiting since future consumers may not use Django.
+
+The strong dependency on Open edX and Django conflicts with our architectural goals of service independence and platform neutrality. Thus, we ruled this option out.
+
+**References**
+
+- `How to start using the Event Bus (Open edX) <https://openedx.atlassian.net/wiki/spaces/AC/pages/3508699151/How+to+start+using+the+Event+Bus>`_
+- `How to Use the Event Bus on edX.org (2U) <https://2u-internal.atlassian.net/wiki/spaces/AT/pages/174555142/How+to+Use+the+Event+Bus+edX.org+2+of+2>`_
+
+2. AWS EventBridge
+==================
+
+AWS EventBridge supports asynchronous service communication through an event bus and API destinations, offering a robust and managed pub/sub system.
+
+**Cons**
+
+- **Operational Cost**: Additional cost for message processing and API destinations.
```

Comment:
Do you have any data on what the cost would be?

Reply:
I asked SRE about it, and they shared this: https://aws.amazon.com/eventbridge/pricing/

Although I am no expert in costing and analysis 😅  but giving a look at the document it seems like the cost is minimal

- Maintaining two API destination connections would cost about $1/month.
- Publishing events would be free for the first 100k events per month, and then $1 per million events beyond that.

Although the main concern for not proposing AWS was related to costs, but since the costs is minimal. we can also go with AWS as a potential solution. 

Could you please provide your opinion on this?